### PR TITLE
[stable/airflow] Allow specifying nodePort for Web service

### PR DESCRIPTION
**Pull request by thirtyseven**
#### What this PR does / why we need it:
Currently, it is not possible to request a specific NodePort in the Airflow chart.  Choosing a NodePort allows it to be included in the Webserver Base URL setting.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/airflow/templates/service-web.yaml [hunk 1]
```diff
@@ -22,4 +22,7 @@ spec:
     - name: web
       protocol: TCP
       port: {{ .Values.airflow.service.externalPort | default 8080 }}
+      {{- if .Values.airflow.service.nodePort }}
```
Comment (Aisuko) [inline stable/airflow/templates/service-web.yaml]: I have a consider on this, why we need to add a NodePort to the chart?   The NodePort should be random create by the controller. Also, the ingress can not transfer the `TCP` traffic, but I think `Istio` can help us do that.

**Review by Aisuko — COMMENTED** [review]
Thanks for your PR and there may some a piece of advice, please have a look on this, thanks.

File: stable/airflow/templates/service-web.yaml (hunk 1 above)
Comment (gsemet) [reply stable/airflow/templates/service-web.yaml]: Hi. I don't know the intend behind this, but freezing the nodeport allows to access to the pod more easily knowning the port, without using an ingress.

I think no matter the use case one should always use an ingress/loadbalancer/whatever, but for debug I can understand the use case.

I would prefer however a solution such as:
```yaml
{{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort.http))) }}
      nodePort: {{ .Values.service.nodePort.http }}
      {{- end }}
```

Comment (Aisuko) [reply stable/airflow/templates/service-web.yaml]: Yes, I believe that can be made more sense.

Comment (gsemet) [reply stable/airflow/templates/service-web.yaml]: @thirtyseven can you update and update your PR pleae?

Comment (thirtyseven) [reply stable/airflow/templates/service-web.yaml]: > Hi. I don't know the intend behind this, but freezing the nodeport allows to access to the pod more easily knowning the port, without using an ingress.
> 
> I think no matter the use case one should always use an ingress/loadbalancer/whatever, but for debug I can understand the use case.

Yes, that's the intent of this PR.  We are using this chart in a k8s environment that does not currently support ingress controllers, but does have Consul DNS for service discovery.  

I'll make the changes as requested.

**Pull request merged**